stream-video-android-core/src/test/kotlin/io/getstream/video/android/core/CallStateTest.kt

```diff
@@ -176,12 +175,12 @@ class CallStateTest : IntegrationTestBase() {
     @Test
     fun `Setting the speaking while muted flag will reset itself after delay`() = runTest {
         // we can make multiple calls, this should have no impact on the reset logic or duration
-        val speakingWhileMuted = call.state.speakingWhileMuted.testIn(backgroundScope)
+        val speakingWhileMuted = call.state.speakingWhileMuted
         call.state.markSpeakingAsMuted()
         call.state.markSpeakingAsMuted()
         call.state.markSpeakingAsMuted()
-        assertTrue(speakingWhileMuted.expectMostRecentItem())
         // The flag should automatically reset to false 2 seconds
-        assertFalse(speakingWhileMuted.awaitItem())
+        advanceTimeBy(2500)
+        assertFalse(speakingWhileMuted.value)
```

Comment:
This is not a good idea - writing a test code based on very static delaying instead of observation, which means if we have much of this code, we will need to wait much time to run unit test codes. We can nicely resolve this with [turbine](https://github.com/cashapp/turbine).

Reply:
This [advanceTimeBy](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/advance-time-by.html) advances the virtual time instantly, so the test actually runs instantly. This makes it possible to test for example timers in our code - I verify that after 2000+ seconds the timer resets the value. With Turbine I would just verify that the value resets, but if the time is correct (it would also pass the test if the value reset after 500ms instead of 2500ms).

Reply:
Sounds sweet! Nicely done.